bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -457,4 +515,74 @@ public static boolean isValidIPConfig(String ipAddress) {
         return false;
     }
 
+    private void scheduleToPollNetworkInterface(int frequencyInSecs) {
+
+        if (networkInterfacePollFuture != null) {
+            networkInterfacePollFuture.cancel(true);
+            networkInterfacePollFuture = null;
+        }
+
+        networkInterfacePollFuture = scheduledExecutorService.scheduleWithFixedDelay(
+                () -> this.pollAndNotifyNetworkInterfaceAddress(), 1, frequencyInSecs, TimeUnit.SECONDS);
+    }
+
+    private void pollAndNotifyNetworkInterfaceAddress() {
+        Collection<CidrAddress> newInterfaceAddresses = getAllInterfaceAddresses();
+        if (networkAddressChangeListeners.isEmpty()) {
+            // no listeners listening, just update
+            lastKnownInterfaceAddresses = newInterfaceAddresses;
+            return;
+        }
+
+        // Look for added addresses to notify
+        List<CidrAddress> added = newInterfaceAddresses.stream()
+                .filter(newInterfaceAddr -> !lastKnownInterfaceAddresses.contains(newInterfaceAddr))
+                .collect(Collectors.toList());
+
+        // Look for removed addresses to notify
+        List<CidrAddress> removed = lastKnownInterfaceAddresses.stream()
+                .filter(lastKnownInterfaceAddr -> !newInterfaceAddresses.contains(lastKnownInterfaceAddr))
+                .collect(Collectors.toList());
+
+        lastKnownInterfaceAddresses = newInterfaceAddresses;
+
+        if (!added.isEmpty() || !removed.isEmpty()) {
+            LOGGER.debug("added {} network interfaces: {}", added.size(), Arrays.deepToString(added.toArray()));
+            LOGGER.debug("removed {} network interfaces: {}", removed.size(), Arrays.deepToString(removed.toArray()));
+
+            notifyListeners(added, removed);
+        }
+    }
+
+    private void notifyListeners(List<CidrAddress> added, List<CidrAddress> removed) {
+        // Prevent listeners changing the list
+        List<CidrAddress> unmodifiableAddedList = Collections.unmodifiableList(added);
+        List<CidrAddress> unmodifiableRemovedList = Collections.unmodifiableList(removed);
+
+        // notify each listener with a timeout of 15 seconds.
+        // SafeCaller prevents bad listeners running too long or throws runtime exceptions
+        networkAddressChangeListeners
+                .forEach(listener -> safeCaller.create(listener, NetworkAddressChangeListener.class).withTimeout(15000)
```

Comment:
As you are blocking up to 15 seconds per listener here, you run the risk that the component has meanwhile been deactivated and the `safeCaller` is null. It might therefore be better, if you use a classical "for" loop and break the execution in case the component has been deactivated meanwhile (i.e. safeCaller is null).

Reply:
very good point, i haven't considered it.